src/sources/kafka.rs

```diff
@@ -56,6 +56,14 @@ enum BuildError {
     KafkaSubscribeError { source: rdkafka::error::KafkaError },
 }
 
+/// Metrics configuration.
+#[configurable_component]
+#[derive(Clone, Debug, Default)]
+struct Metrics {
+    /// Expose topic lag metrics for all topics and partitions. The metric name is `kafka_consumer_lag`.
+    pub topic_lag_metric: bool,
```

Comment:
nit: should this be plural since there will often be many metrics generated from this flag?

Reply:
Fixed.